lib/fluent/plugin/exception_detector.rb

```diff
@@ -41,21 +41,26 @@ def state
       end
     end
 
-    def self.rule(from_state, pattern, to_state)
-      Struct::Rule.new(from_state, pattern, to_state)
+    def self.rule(from_states, pattern, to_state)
```

Comment:
thoughts on the interface of this function:
can we always assume from_states is `Array`, and clients always need to pass Array in? 2 benefits:
1. no special logic to convert single element to an array
2. the client side code are consistent.

Reply:
Instead of thinking of this as a function, think of this as a DSL for defining rules. We want to be able to define rules with a single transition, or with multiple transitions that share a regexp and the target state. In the former case we specify a single from state, and in the latter we specify an array of from states. Most rules will represent a single transition. The goal is to make the usage simpler, not the function implementation.

Reply:
[Optional] 
1. still think `from_states` implies input of this parameter is a collection of items.
2. still think always pass `from_states` parameter as array makes the client side code consistent - so user of this function (or DSL) don't need to reason two modes to use this `rule` function.

This comment falls under readability realm, so it's optional to change.

Reply:
Good point on the variable name ­— renamed to `from_state_or_states`.
The idea was to keep most of the existing `rule` statements unchanged.